Question 1: Sneakers
Please see code and explanation at the end
Question 2: End of notebook

In [1]:
import pandas as pd

In [3]:
file_path = '/home/andy/Documents/Shopify/2019 Winter Data Science Intern Challenge Data Set - Sheet1.csv'
df = pd.read_csv(file_path)
df.head()

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0         1       53      746           224            2           cash   
1         2       92      925            90            1           cash   
2         3       44      861           144            1           cash   
3         4       18      935           156            1    credit_card   
4         5       18      883           156            1    credit_card   

            created_at  
0  2017-03-13 12:36:56  
1  2017-03-03 17:38:52  
2   2017-03-14 4:23:56  
3  2017-03-26 12:43:37  
4   2017-03-01 4:35:11

In [13]:
df.describe()

order_id      shop_id      user_id   order_amount  total_items
count  5000.000000  5000.000000  5000.000000    5000.000000   5000.00000
mean   2500.500000    50.078800   849.092400    3145.128000      8.78720
std    1443.520003    29.006118    87.798982   41282.539349    116.32032
min       1.000000     1.000000   607.000000      90.000000      1.00000
25%    1250.750000    24.000000   775.000000     163.000000      1.00000
50%    2500.500000    50.000000   849.000000     284.000000      2.00000
75%    3750.250000    75.000000   925.000000     390.000000      3.00000
max    5000.000000   100.000000   999.000000  704000.000000   2000.00000

In [28]:
df['order_amount'].mean()

3145.128

In [60]:
max_monthly_purchases = 30

df_business = df[df['shop_id'].isin([42,78])]
df_consumers = df[~df['shop_id'].isin([42,78])]

print("Average Shoe Sold of Consumers: ", df_consumers['order_amount'].sum() / df_consumers['total_items'].sum())
print("Average Shoe Sold of Businesses: ", df_business['order_amount'].sum() / df_business['total_items'].sum())
print("Average Shoe of All Customers: ", df['order_amount'].sum() / df['total_items'].sum())
print()
print("Average Order of Consumers: ", df_consumers['order_amount'].mean())
print("Average Order of Businesses: ", df_business[df_business['user_id'] != 607]['order_amount'].mean())
print("Average Order of Business Customer 607: ", df_business[df_business['user_id'] == 607]['order_amount'].mean())
print("Average Order of All Customers: ", df['order_amount'].mean())

Average Shoe Sold of Consumers:  150.4
Average Shoe Sold of Businesses:  417.38092588796815
Average Shoe of All Customers:  357.92152221412965

Average Order of Consumers:  300.1558229655313
Average Order of Businesses:  28574.7
Average Order of Business Customer 607:  704000.0
Average Order of All Customers:  3145.128


A) What went wrong? 
There's clearly 2 distinct sets of customers which intuitively are consumers and businesses with store IDs 78 & 42 being the businesses selling large order values for small order quantities (and being the answer to everything else in the universe.)
There is also a power user skewing data, user ID 607 that needs to be accounted for. 
Thus consumers order amount corresponds to a shoe whereas business order amount seems to correspond to batches or be mixed with some batches and some corresponding to expensive shoes. 


B) What metric would I report for this dataset?

For consumers the average shoe price or order size is appropriate and for businesses average order price is appropriate because order quantities are ambiguous between shoes / batch. 
The data is also being skewed by user_id 607 who buys a large amount of shoes from shop 42.

Aside: 
The underlying goal seems to be to determine the types of consumers that shop at each store so more insights can be derived from clustering algorithms given the attributes.
We can see if there are any relationships to dates (perhaps sneaker drops or holidays), power users / sneaker heads, and shop popularity.

However, assuming we just want to look at general affordability, the metric I would report is AOV 

C)
As such:
Avg Shoe Order of Consumers: $150.40 / shoe
Avg Order Size of Consumers: $300.15 / order
Avg Order Size of Businesses: $28,574.00 / order
Avg Order Size of User 607: $704,000.00 /order



Question 2:
A) 
SQL Query: 

SELECT DISTINCT COUNT(OrderID) FROM Orders
INNER JOIN Shippers
ON Orders.ShipperID = Shippers.ShipperID
WHERE ShipperName = "Speedy Express"
;

Result: 54

B) 
SELECT TOP 1 LastName FROM(
SELECT DISTINCT COUNT(OrderID) AS QTY, LastName FROM Orders
INNER JOIN Employees
ON Orders.EmployeeID = Employees.EmployeeID
GROUP BY LastName
ORDER BY 1 DESC
)
;

Result: Peacock

C)
SELECT COUNT(OrderDetails.OrderID), Products.ProductName, SUM(Quantity) FROM (OrderDetails
INNER JOIN Orders
ON OrderDetails.OrderID = Orders.OrderID)
INNER JOIN Customers
ON Orders.CustomerID = Customers.CustomerID
INNER JOIN Products
ON OrderDetails.ProductID = Products.ProductID
WHERE
Country = 'Germany'
GROUP BY ProductName
ORDER BY 3 DESC
;

Result:
QTY: Boston Boston Crab Meat
Order Freq: Gorgonzola Telino